idaes/generic_models/properties/core/examples/tests/test_CO2_H2O_MEA_Ideal_VLE.py

```diff
@@ -0,0 +1,273 @@
+        assert model.props[1].default_energy_balance_type() == \
+            EnergyBalanceType.enthalpyTotal
+
+    @pytest.mark.unit
+    def test_get_material_flow_basis(self, model):
+        assert model.props[1].get_material_flow_basis() == \
+            MaterialFlowBasis.molar
+
+    @pytest.mark.unit
+    def test_define_state_vars(self, model):
+        sv = model.props[1].define_state_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["flow_mol",
+                         "mole_frac_comp",
+                         "temperature",
+                         "pressure"]
+
+    @pytest.mark.unit
+    def test_define_port_members(self, model):
+        sv = model.props[1].define_state_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["flow_mol",
+                         "mole_frac_comp",
+                         "temperature",
+                         "pressure"]
+
+    @pytest.mark.unit
+    def test_define_display_vars(self, model):
+        sv = model.props[1].define_display_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["Total Molar Flowrate",
+                         "Total Mole Fraction",
+                         "Temperature",
+                         "Pressure"]
+
+    @pytest.mark.unit
+    def test_dof(self, model):
+        # Fix state
+        model.props[1].flow_mol.fix(92.0817)
+        model.props[1].temperature.fix(380)
+        model.props[1].pressure.fix(183430)
+        model.props[1].mole_frac_comp["H2O"].fix(0.8787745)
+        model.props[1].mole_frac_comp["CO2"].fix(0.018288)
+        model.props[1].mole_frac_comp["MEA"].fix(0.1029375)
+
+        assert degrees_of_freedom(model.props[1]) == 0
+
+    @pytest.mark.initialize
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, model):
+        orig_fixed_vars = fixed_variables_set(model)
+        orig_act_consts = activated_constraints_set(model)
+
+        model.props.initialize(optarg={'tol': 1e-6})
+
+        assert degrees_of_freedom(model) == 0
+
+        fin_fixed_vars = fixed_variables_set(model)
+        fin_act_consts = activated_constraints_set(model)
+
+        assert len(fin_act_consts) == len(orig_act_consts)
+        assert len(fin_fixed_vars) == len(orig_fixed_vars)
+
+        for c in fin_act_consts:
+            assert c in orig_act_consts
+        for v in fin_fixed_vars:
+            assert v in orig_fixed_vars
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, model):
+        results = solver.solve(model)
+
+        # Check for optimal solution
+        assert results.solver.termination_condition == \
+            TerminationCondition.optimal
+        assert results.solver.status == SolverStatus.ok
```

Comment:
You need to compare the model results to something from literature, otherwise we have no way of knowing if the model is right.